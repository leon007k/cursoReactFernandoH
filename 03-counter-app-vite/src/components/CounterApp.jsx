import PropTypes from 'prop-types';
import { useState } from 'react';

export function CounterApp({ value }) {

  const [counter, setCounter] = useState(value);

  const handleClickIncrementCounter = () => setCounter(counter + 1)

  const handleClickDecrementCounter = () => setCounter(counter - 1);

  const handleClickResetCounter = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2 id='heading-increment'>Incrementa: {counter}</h2>
      <p>Toca el boton, para incrementar el valor </p>

      <button aria-label='btn-increment' onClick={handleClickIncrementCounter}> +1 </button>
      <button aria-label='btn-decrement' onClick={handleClickDecrementCounter}> -1 </button>
      <button aria-label='btn-reset' onClick={handleClickResetCounter}> Reset </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}