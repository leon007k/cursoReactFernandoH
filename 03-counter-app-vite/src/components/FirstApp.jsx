import PropTypes from "prop-types";
import { ListOfMessages } from "../const";

function getListOfMessages(id) {
  const messages = ListOfMessages.find(message => message.id === id);
  return { autorMessage: messages.name, messages: messages.message };
}

export default function FirstApp({ title = 'Hello World from FirstApp', subtitle = 'This is a subtitle' }) {

  const { autorMessage, messages } = getListOfMessages(2);

  return (
    <>
      <h2 data-testid="test-title">{title}</h2>
      <h3>{subtitle}</h3>
      <h3>{subtitle}</h3>
      <p>
        Autor: {autorMessage}
        <br />
        Mensaje: {messages}
      </p>

    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

/*
  * Se remueven por el warning:
  ! FirstApp: Support for defaultProps will be removed from function components in a future major release.
  ! Use JavaScript default parameters instead
FirstApp.defaultProps = {
  title: 'Hello World from FirstApp',
  subtitle: 'This is a subtitle'
} */
