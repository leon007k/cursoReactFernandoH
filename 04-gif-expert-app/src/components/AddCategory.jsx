import { useState } from 'react'
import { PropTypes } from 'prop-types'

export function AddCategory({ onNewCategories }) {

  const [inputValue, setInputValue] = useState('')

  const handleInputChange = ({ target }) => {
    setInputValue(target.value)
  }

  const handleOnSubmit = (evt) => {
    evt.preventDefault()
    const cleanCategory = inputValue.trim()
    if (cleanCategory.length <= 1) return

    onNewCategories(cleanCategory)
    setInputValue('')
  }

  return (
    <form onSubmit={handleOnSubmit} aria-label='form'>
      <input
        type="text"
        placeholder="Agregar categoria"
        value={inputValue}
        onChange={handleInputChange}
      />
      <button className='btn' onClick={handleOnSubmit}>Agregar</button>
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategories: PropTypes.func.isRequired,
}