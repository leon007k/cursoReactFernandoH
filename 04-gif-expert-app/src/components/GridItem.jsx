import { PropTypes } from 'prop-types'
import '../style.css'

export function GridItem({ title, url }) {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}