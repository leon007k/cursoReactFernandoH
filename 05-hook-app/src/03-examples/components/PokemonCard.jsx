export  function PokemonCard({id, name, sprites = []}) {
  return (
    <section style={{ height: 200}} className="card text-center p-3">
      <h2 className="text-capitalize">#{id} - {name}</h2>

      {/** Imagenes */}
      <div>
        {
          sprites.map( sprite => (
            <img key={sprite} src={sprite} alt={name}/>
          ))
        }
      </div>
    </section>
  )
}
