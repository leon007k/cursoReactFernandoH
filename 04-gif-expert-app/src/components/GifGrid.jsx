import { PropTypes } from 'prop-types'

import { GridItem } from './GridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'
import '../style.css'

export function GifGrid({ category }) {

  const { images, isLoading } = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h4>Cargando...</h4>}
      <div className='card-grid'>
        {
          images.map((image) => (
            <GridItem key={image.id} {...image} />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}