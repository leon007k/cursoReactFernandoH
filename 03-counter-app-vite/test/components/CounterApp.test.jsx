import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../../src/components/CounterApp';

describe('Pruebas en <CounterApp />', () => {

  const initialValue = 100

  test('Debe hacer match con el snapshot', () => {

    const { container } = render(<CounterApp value={initialValue} />)

    expect(container).toMatchSnapshot()
  })

  test('Debe de mostrar el valor inicial en 100', () => {

    const incrementMsg = `Incrementa: ${initialValue}`

    render(<CounterApp value={initialValue} />)

    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toBe(incrementMsg)
  })

  test('Debe de incrementar con el botón +1', () => {

    render(<CounterApp value={initialValue} />)

    // * fireEvent -> simula un evento en un elemento
    fireEvent.click(screen.getByText('+1'))
    expect(screen.getByText(`Incrementa: ${initialValue + 1}`)).toBeTruthy()
  })

  test('Debe de decrementar con el botón -1', () => {

    render(<CounterApp value={initialValue} />)

    // * fireEvent -> simula un evento en un elemento
    fireEvent.click(screen.getByText('-1'))
    expect(screen.getByText(`Incrementa: ${initialValue - 1}`)).toBeTruthy()
  })

  test('Debe de funcionar el boton reset', () => {

    render(<CounterApp value={initialValue} />)

    fireEvent.click(screen.getByRole('button', { name: 'btn-increment' }))
    fireEvent.click(screen.getByRole('button', { name: 'btn-increment' }))
    fireEvent.click(screen.getByRole('button', { name: 'btn-increment' }))

    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }))

    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toBe(`Incrementa: ${initialValue}`)

  })
});