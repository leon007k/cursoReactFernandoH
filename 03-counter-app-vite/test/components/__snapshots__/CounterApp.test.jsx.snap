// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    id="heading-increment"
  >
    Incrementa: 
    100
  </h2>
  <p>
    Toca el boton, para incrementar el valor 
  </p>
  <button
    aria-label="btn-increment"
  >
     +1 
  </button>
  <button
    aria-label="btn-decrement"
  >
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
